### ISSUE TITLE
Expected: 'Nothing else in FileSegment.' with multiple `GRANT` statements in one file (postgres dialect)

### ISSUE THREAD

**Issue by kquinsland** [sqlfluff/sqlfluff#7238]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

This appears to be similar to https://github.com/sqlfluff/sqlfluff/issues/5330.

Given `test.sql`:

```sql
\c example
BEGIN;

CREATE ROLE some_role;
ALTER ROLE some_role WITH LOGIN INHERIT PASSWORD 'SecretSquirrel';

GRANT somegeneralpolicy TO some_role;

GRANT SELECT ON
  someTable,
  someOtherTable,
TO some_role

COMMIT;
```

I am getting a lint error: `Found unparsable section:`

### Expected Behaviour

I would expect this to at least be parsable

### Observed Behaviour

```shell
❯ sqlfluff lint --dialect postgres schemas/test.sql
== [schemas/test.sql] FAIL                                                                                                                                       
L:   9 | P:   1 |  PRS | Line 9, Position 1: Found unparsable section: 'GRANT
                       | SELECT ON\n  someTable,\n  someOther...'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```shell
❯ sqlfluff parse schemas/test.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    meta_command:                                             '\\c example'
[L:  1, P: 11]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    statement:
[L:  2, P:  1]      |        transaction_statement:
[L:  2, P:  1]      |            keyword:                                          'BEGIN'
[L:  2, P:  6]      |    statement_terminator:                                     ';'
[L:  2, P:  7]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    newline:                                                  '\n'
[L:  4, P:  1]      |    statement:
[L:  4, P:  1]      |        create_role_statement:
[L:  4, P:  1]      |            keyword:                                          'CREATE'
[L:  4, P:  7]      |            whitespace:                                       ' '
[L:  4, P:  8]      |            keyword:                                          'ROLE'
[L:  4, P: 12]      |            whitespace:                                       ' '
[L:  4, P: 13]      |            role_reference:
[L:  4, P: 13]      |                naked_identifier:                             'some_role'
[L:  4, P: 22]      |    statement_terminator:                                     ';'
[L:  4, P: 23]      |    newline:                                                  '\n'
[L:  5, P:  1]      |    statement:
[L:  5, P:  1]      |        alter_role_statement:
[L:  5, P:  1]      |            keyword:                                          'ALTER'
[L:  5, P:  6]      |            whitespace:                                       ' '
[L:  5, P:  7]      |            keyword:                                          'ROLE'
[L:  5, P: 11]      |            whitespace:                                       ' '
[L:  5, P: 12]      |            role_reference:
[L:  5, P: 12]      |                naked_identifier:                             'some_role'
[L:  5, P: 21]      |            whitespace:                                       ' '
[L:  5, P: 22]      |            keyword:                                          'WITH'
[L:  5, P: 26]      |            whitespace:                                       ' '
[L:  5, P: 27]      |            keyword:                                          'LOGIN'
[L:  5, P: 32]      |            whitespace:                                       ' '
[L:  5, P: 33]      |            keyword:                                          'INHERIT'
[L:  5, P: 40]      |            whitespace:                                       ' '
[L:  5, P: 41]      |            keyword:                                          'PASSWORD'
[L:  5, P: 49]      |            whitespace:                                       ' '
[L:  5, P: 50]      |            quoted_literal:                                   "'SecretSquirrel'"
[L:  5, P: 66]      |    statement_terminator:                                     ';'
[L:  5, P: 67]      |    newline:                                                  '\n'
[L:  6, P:  1]      |    newline:                                                  '\n'
[L:  7, P:  1]      |    statement:
[L:  7, P:  1]      |        access_statement:
[L:  7, P:  1]      |            keyword:                                          'GRANT'
[L:  7, P:  6]      |            whitespace:                                       ' '
[L:  7, P:  7]      |            object_reference:
[L:  7, P:  7]      |                naked_identifier:                             'somegeneralpolicy'
[L:  7, P: 24]      |            whitespace:                                       ' '
[L:  7, P: 25]      |            keyword:                                          'TO'
[L:  7, P: 27]      |            whitespace:                                       ' '
[L:  7, P: 28]      |            role_reference:
[L:  7, P: 28]      |                naked_identifier:                             'some_role'
[L:  7, P: 37]      |    statement_terminator:                                     ';'
[L:  7, P: 38]      |    newline:                                                  '\n'
[L:  8, P:  1]      |    newline:                                                  '\n'
[L:  9, P:  1]      |    unparsable:                                               !! Expected: 'Nothing else in FileSegment.'
[L:  9, P:  1]      |        word:                                                 'GRANT'
[L:  9, P:  6]      |        whitespace:                                           ' '
[L:  9, P:  7]      |        word:                                                 'SELECT'
[L:  9, P: 13]      |        whitespace:                                           ' '
[L:  9, P: 14]      |        word:                                                 'ON'
[L:  9, P: 16]      |        newline:                                              '\n'
[L: 10, P:  1]      |        whitespace:                                           '  '
[L: 10, P:  3]      |        word:                                                 'someTable'
[L: 10, P: 12]      |        comma:                                                ','
[L: 10, P: 13]      |        newline:                                              '\n'
[L: 11, P:  1]      |        whitespace:                                           '  '
[L: 11, P:  3]      |        word:                                                 'someOtherTable'
[L: 11, P: 17]      |        comma:                                                ','
[L: 11, P: 18]      |        newline:                                              '\n'
[L: 12, P:  1]      |        word:                                                 'TO'
[L: 12, P:  3]      |        whitespace:                                           ' '
[L: 12, P:  4]      |        word:                                                 'some_role'
[L: 12, P: 13]      |        newline:                                              '\n'
[L: 13, P:  1]      |        newline:                                              '\n'
[L: 14, P:  1]      |        word:                                                 'COMMIT'
[L: 14, P:  7]      |        semicolon:                                            ';'
[L: 14, P:  8]      |    newline:                                                  '\n'
[L: 15, P:  1]      |    [META] end_of_file:

==== parsing violations ====
L:   9 | P:   1 |  PRS | Line 9, Position 1: Found unparsable section: 'GRANT SELECT ON\n 
                       | someTable,\n  someOther...'
WARNING: Parsing errors found and dialect is set to 'postgres'. Have you configured your dialect correctly?
```

### Dialect

postgres

### Version

```shell
❯ which sqlfluff
/opt/homebrew/bin/sqlfluff
❯ sqlfluff version
3.5.0
❯ python3 --version
Python 3.13.9
```

### Configuration

```toml
# See: https://docs.sqlfluff.com/en/stable/configuration/setting_configuration.html
# Format is python cfg: https://docs.python.org/3/library/configparser.html
##
[sqlfluff]
dialect = postgres
max_line_length = 150

[rules:capitalisation.keywords]
capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by WittierDinosaur** [comment]
aren't you missing a statement terminator between the grant statement and the commit?

**Comment by kquinsland01** [comment]
> aren't you missing a statement terminator between the grant statement and the commit?

Good catch! (also, sorry its taken me so long to get time to get back to you!)

I gave that a shot and ... yes, that was it, effectively. 

Given a simpler test case:

```shell
\c example

BEGIN;

GRANT SELECT ON someTable, someOtherTable TO some_role;

COMMIT;
```

The return is clean:

```
❯ sqlfluff parse --dialect postgres schemas/simple-test.sql
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    meta_command:                                             '\\c example'
[L:  1, P: 11]      |    newline:                                                  '\n'
[L:  2, P:  1]      |    newline:                                                  '\n'
[L:  3, P:  1]      |    statement:
[L:  3, P:  1]      |        transaction_statement:
[L:  3, P:  1]      |            keyword:                                          'BEGIN'
[L:  3, P:  6]      |    statement_terminator:                                     ';'
[L:  3, P:  7]      |    newline:                                                  '\n'
[L:  4, P:  1]      |    newline:                                                  '\n'
[L:  5, P:  1]      |    statement:
[L:  5, P:  1]      |        access_statement:
[L:  5, P:  1]      |            keyword:                                          'GRANT'
[L:  5, P:  6]      |            whitespace:                                       ' '
[L:  5, P:  7]      |            keyword:                                          'SELECT'
[L:  5, P: 13]      |            whitespace:                                       ' '
[L:  5, P: 14]      |            keyword:                                          'ON'
[L:  5, P: 16]      |            whitespace:                                       ' '
[L:  5, P: 17]      |            object_reference:
[L:  5, P: 17]      |                naked_identifier:                             'SOMETABLE'
[L:  5, P: 26]      |            comma:                                            ','
[L:  5, P: 27]      |            whitespace:                                       ' '
[L:  5, P: 28]      |            object_reference:
[L:  5, P: 28]      |                naked_identifier:                             'SOMEOTHERTABLE'
[L:  5, P: 42]      |            whitespace:                                       ' '
[L:  5, P: 43]      |            keyword:                                          'TO'
[L:  5, P: 45]      |            whitespace:                                       ' '
[L:  5, P: 46]      |            role_reference:
[L:  5, P: 46]      |                naked_identifier:                             'SOME_ROLE'
[L:  5, P: 55]      |    statement_terminator:                                     ';'
[L:  5, P: 56]      |    newline:                                                  '\n'
[L:  6, P:  1]      |    newline:                                                  '\n'
[L:  7, P:  1]      |    statement:
[L:  7, P:  1]      |        transaction_statement:
[L:  7, P:  1]      |            keyword:                                          'COMMIT'
[L:  7, P:  7]      |    statement_terminator:                                     ';'
[L:  7, P:  8]      |    newline:                                                  '\n'
[L:  8, P:  1]      |    [META] end_of_file:
```
